ENH: Updates to cluster_features_de_novo

gregcaporaso opened the pull request:
Addressing #12 will improve the test coverage, but we can do that in another PR since the `.uc` parser will likely change a little bit. We should be able to use one `.uc` parser for all of the methods we're currently planning on implementing.

thermokarst commented:
@gregcaporaso - your commit messages look like this PR should be rebased, not squashed, when approved. Can you confirm when you get a chance? Thanks!

thermokarst commented on q2_vsearch/tests/test_cluster_features.py:
```diff
@@ -0,0 +1,187 @@
+                                      constructor=skbio.DNA, format='fasta'))
+        exp_seqs = list(skbio.io.read(str(input_sequences),
+                                      constructor=skbio.DNA, format='fasta'))
+        print(obs_seqs)
```
Can you drop this please?

thermokarst replied:
`print(gerg)`

thermokarst commented on q2_vsearch/tests/test_cluster_features.py:
```diff
@@ -0,0 +1,187 @@
+        exp_seqs = list(skbio.io.read(str(input_sequences),
+                                      constructor=skbio.DNA, format='fasta'))
+        print(obs_seqs)
+        print(exp_seqs)
```
Can you drop this please?

thermokarst commented on q2_vsearch/_cluster_features.py:
```diff
@@ -33,29 +43,38 @@ def _collapse_f_from_uc(uc):
+                        skbio.io.read(sequences_fp, constructor=skbio.DNA,
+                                      format='fasta')}
+    feature_ids_table = set(table.ids(axis='observation'))
+    non_overlapping_ids = feature_ids_seqs ^ feature_ids_table
```
👍

thermokarst commented on q2_vsearch/plugin_setup.py:
```diff
@@ -17,23 +25,42 @@
+        'sequences': 'The sequences corresponding to the features in table.',
+    },
+    parameter_descriptions={
+        'id': 'The percent identity at which clustering should be performed.',
```
I know the VSEARCH flag is `--id`, but how would you feel about `identity` here? I keep seeing this and thinking of "identifier" instead of "identity." It isn't a showstopper for me, just thought I would ask.

gregcaporaso replied:
I went back and forth on that... I ended up going with `id` to match the vsearch parameter name. In this case it's probably fine to not match vsearch, but in general I was thinking matching vsearch's parameter names might be a useful strategy, so users can easily refer to the vsearch documentation. It's really likely that we're going to expose other vsearch parameters as well, so it will help to easily be able to find info in their docs. 

I'm fine either way for this one - you call it @thermokarst.

thermokarst replied:
Just chatted IRL with @jairideout about this - maybe it makes sense to match the `perc-identity` parameter in [`classify-consensus-vsearch`](https://docs.qiime2.org/2017.8/plugins/available/feature-classifier/classify_consensus_vsearch/) and [`classify-consensus-blast`](https://docs.qiime2.org/2017.8/plugins/available/feature-classifier/classify_consensus_blast/), and just including a note in the `description` saying something like "This parameter maps to the `--id` flag in `vsearch`"

gregcaporaso commented:
@thermokarst, this addresses your comments minus the name change to the `id` parameter (see my comment above - let me know if you feel strongly about changing its name). All commits can be squashed.

gregcaporaso commented:
@thermokarst, this one should be good to go.
